bin/database.py

```
@@ -25,19 +25,44 @@
 from api.types import Origin
 from api.jobs import batch
 
+import fixes
+from process_cursor import process_cursor
 
 CURRENT_DATABASE_VERSION = 62 # An int that is bumped when a new schema change is made
 
 
 def get_db_version():
+    """Get the current db version, with applied fixes.
 
+    Returns:
+        (int, dict): The current database version, and a map of applied fixes
+    """
     version = config.get_version()
     if version is None:
         # Attempt to find db version at old location
         version = config.db.version.find_one({'_id': 'version'})
     if version is None or version.get('database') is None:
         return 0
-    return version.get('database')
+    return version.get('database'), version.get('applied_fixes', {})
+
+def get_available_fixes(db_version, applied_fixes):
+    """Get a list of fixes that should be applied for the given db version.
+
+    Args:
+        db_version (int): The current db version
+        applied_fixes (dict): The set of fixes that have already been applied
+
+    Returns:
+        list(str): The list of fix_ids to be applied
+    """
+    available_fixes = fixes.AVAILABLE_FIXES.get(db_version, [])
+
+    result = []
+    for fix in available_fixes:
+        if fix not in applied_fixes:
+            result.append(fix)
+
+    return result
```

This function should probably error out if a fix is not present in the DB, for a lower version that the DB is currently at. Or if not here, definitely somewhere - I don't see right now where that would happen if one is skipped.